### PR TITLE
Fixup test for RANSAC: don't pick duplicate samples #3901

### PR DESCRIPTION
**Pull request by hmaarrfk**
## Description

Seems that residuals needs to be the size of data.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by jni** [comment]
Why was it working before?

**Comment by hmaarrfk** [comment]
This totally has to to with a numpy 1.12 only bug. 

```
E               IndexError: in the future, 0-d boolean arrays will be interpreted as a valid boolean index
```

This error was found when I trigged the builds in the wheels, some of which are built with numpy 1.12

https://travis-ci.org/scikit-image/scikit-image-wheels/builds/533603305?utm_source=github_status&utm_medium=notification

I guess this is somewhere where dynamically typed languages fail to help us identify errors at build time.

File: skimage/measure/tests/test_fit.py [hunk 1]
```diff
@@ -361,7 +361,7 @@ def estimate(self, data):
             return True
 
         def residuals(self, data):
-            return 1.0
+            return np.ones_like(data, dtype=np.double)
```
Comment (jni) [inline skimage/measure/tests/test_fit.py]: shouldn't this be `np.ones(len(data), dtype=np.double)`?

**Comment by hmaarrfk** [comment]
The test was mostly designed to test for duplicates, but didn't really ensure that residuals was the correct return

These 3 are the offending lines you should look at
https://github.com/scikit-image/scikit-image/blob/master/skimage/measure/fit.py#L832
https://github.com/scikit-image/scikit-image/blob/master/skimage/measure/fit.py#L847
https://github.com/scikit-image/scikit-image/blob/master/skimage/measure/fit.py#L861
This is the line in particular

File: skimage/measure/tests/test_fit.py (hunk 1 above)
Comment (hmaarrfk) [reply skimage/measure/tests/test_fit.py]: ohhh, yeah you are right.

Comment (hmaarrfk) [inline skimage/measure/tests/test_fit.py]:
```suggestion
            return np.ones(len(data), dtype=np.double)
```

**Comment by hmaarrfk** [comment]
@meeseeksdev backport to v0.14.x

**Comment by hmaarrfk** [comment]
Merging was done pre-emptively to speed up the release process of 0.14.3 after local testing was made to ensure relevant tests were fixed.

**Pull request merged**